### PR TITLE
feat(rich-text-editor): add popup for value with variables for link/image

### PR DESCRIPTION
**Pull request by stanleynguyen**
## Problem

There's no way for users to input variables inside link hrefs or img src easily right now

## Solution

A modal with input that has variable suggestion to input value for links hrefs and img src

## Tests

- [x] use variables inside link href
- [x] use variables inside img src

**Know drawbacks**:  Images won't render correctly inside the rich text editor as the actual value is not yet being used

## Deploy Notes

- [ ] Migrate all `sendEmailWithOldEditor` to the new one

### CODE REVIEW

File: packages/frontend/src/components/RichTextEditor/MenuBar.tsx [hunk 1]
```diff
@@ -19,23 +20,56 @@ import {
   RiTableLine,
   RiUnderline,
 } from 'react-icons/ri'
+import { LoadingButton } from '@mui/lab'
+import {
+  Dialog,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from '@mui/material'
```
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/MenuBar.tsx:29]: is it possible to use chakra for this? cus we wanna move out of mui right

File: packages/frontend/src/components/RichTextEditor/MenuBar.tsx [hunk 2]
```diff
@@ -190,35 +224,147 @@ interface MenuBarProps {
...
+        return
+      }
+      editor.chain().focus().setImage({ src: url }).run()
+      setShowValueDialog(false)
+    }, [editor, dialogValue]),
+  }
+  const dialogPlaceholders: { [key: string]: string } = {
+    [MenuLabels.LinkSet]: 'Enter a full URL with http prefix',
+    [MenuLabels.ImageAdd]:
+      'Enter direct image link (e.g. https://file.go.gov.sg/clipplumer.png)',
```
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/MenuBar.tsx]: nit: typo here? or without the letter b on purpose hmmmm

File: packages/backend/src/apps/postman/actions/index.ts [hunk 3]
```diff
@@ -1,9 +1,4 @@
-import sendEmailWithOldEditor from './send-email-raw-html'
 import sendEmailWithAttachments from './send-email-with-attachments'
 import sendTransactionalEmail from './send-transactional-email'
 
-export default [
-  sendTransactionalEmail,
-  sendEmailWithAttachments,
-  sendEmailWithOldEditor,
-]
+export default [sendTransactionalEmail, sendEmailWithAttachments]
```
Comment (m0nggh) [inline packages/backend/src/apps/postman/actions/index.ts:4]: should we still keep the old editor and remove it in another PR until like 1-2 weeks later?

File: packages/frontend/src/components/RichTextEditor/MenuBar.tsx [hunk 4]
```diff
@@ -190,35 +224,147 @@ interface MenuBarProps {
...
+              <Form onSubmit={() => dialogOnSubmits[dialogLabel]()}>
+                <BareEditor
+                  // val is in HTML, need to parse back to plain text
+                  onChange={(val) => setDialogValue(parse(val).textContent)}
+                  initialValue={dialogValue}
+                  editable
+                  variablesEnabled
+                  placeholder={dialogPlaceholders[dialogLabel]}
+                />
+                <LoadingButton
```
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/MenuBar.tsx:357]: is it possible to disable if the content is empty? ignore if its too difficult or else can just tell user in the `dialogPlaceholder` that empty link will unset the url

**Review by m0nggh — CHANGES_REQUESTED** [review]
Tested that it works for both link and image with variables in the dialog, amazing work!
Just needed to address a UI issue where the `SuggestionPopper` is not showing at the bottom like I can only scroll to use the first few variables:
image

File: packages/frontend/src/components/RichTextEditor/MenuBar.tsx (hunk 1 above)
Comment (stanleynguyen) [reply packages/frontend/src/components/RichTextEditor/MenuBar.tsx:29]: Hmm we don't have a modal component using chakra yet leh, I think let's solve it in another PR

File: packages/frontend/src/components/RichTextEditor/MenuBar.tsx (hunk 2 above)
Comment (stanleynguyen) [reply packages/frontend/src/components/RichTextEditor/MenuBar.tsx]: lol ok i go fix the link. It's an actual link I typo-ed

File: packages/backend/src/apps/postman/actions/index.ts (hunk 3 above)
Comment (stanleynguyen) [reply packages/backend/src/apps/postman/actions/index.ts:4]: ah this one no need, I don't see a scenario why we need to revert

File: packages/frontend/src/components/RichTextEditor/MenuBar.tsx (hunk 4 above)
Comment (stanleynguyen) [reply packages/frontend/src/components/RichTextEditor/MenuBar.tsx:357]: I thought about this, having empty as unset makes sense to me eh 🤔 that's why I kept it this way

**Review by m0nggh — APPROVED** [review]
ok as discussed with @stanleynguyen, the PR for migrating to chakra model will be done in another PR (thksss), lgtm!

**Pull request merged**